dotnet/src/Connectors/Connectors.AssemblyAI/AudioToText/AssemblyAIAudioToTextService.cs

```diff
@@ -0,0 +1,292 @@
+        await ThrowIfNotSuccessStatusCodeAsync("Failed to upload file.", response, ct)
+            .ConfigureAwait(false);
+        var jsonStream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
+        var json = await JsonDocument.ParseAsync(jsonStream, cancellationToken: ct).ConfigureAwait(false);
+        return json.RootElement.GetProperty("upload_url").GetString()
+               ?? throw new AssemblyAIApiException("Property 'upload_url' expected but not found.");
+    }
+
+    private async Task<string> CreateTranscriptAsync(
+        string audioUrl,
+        PromptExecutionSettings? executionSettings = null,
+        CancellationToken ct = default
+    )
+    {
+        const string URL = "https://api.assemblyai.com/v2/transcript";
+        var jsonRequest = new JsonObject();
+        jsonRequest["audio_url"] = audioUrl;
+        if (executionSettings?.ExtensionData is not null)
+        {
+            foreach (var attribute in executionSettings.ExtensionData)
+            {
+                jsonRequest[attribute.Key] = JsonValue.Create(attribute.Value);
+            }
+        }
+
+        using var content = new StringContent(jsonRequest.ToJsonString(), Encoding.UTF8, "application/json");
+        using var request = new HttpRequestMessage(HttpMethod.Post, URL);
+        request.Headers.Authorization = new AuthenticationHeaderValue(this._apiKey);
+        request.Content = content;
+        using var response = await this._httpClient.SendAsync(request, ct).ConfigureAwait(false);
+        await ThrowIfNotSuccessStatusCodeAsync("Failed to create transcript", response, ct)
+            .ConfigureAwait(false);
+        var jsonStream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
+        var json = await JsonDocument.ParseAsync(jsonStream, cancellationToken: ct).ConfigureAwait(false);
+        if (json.RootElement.TryGetProperty("error", out var property))
+        {
+            throw new AssemblyAIApiException($"Failed to create transcript. Reason: {property.GetString()!}");
+        }
+
+        return json.RootElement.GetProperty("id").GetString()!;
+    }
+
+    private async Task<JsonDocument> WaitForTranscriptToProcessAsync(string transcriptId, CancellationToken ct)
+    {
+        var url = $"https://api.assemblyai.com/v2/transcript/{transcriptId}";
+
+        while (!ct.IsCancellationRequested)
+        {
+            using var request = new HttpRequestMessage(HttpMethod.Get, url);
+            request.Headers.Authorization = new AuthenticationHeaderValue(this._apiKey);
+            using var response = await this._httpClient.SendAsync(request, ct).ConfigureAwait(false);
+            await ThrowIfNotSuccessStatusCodeAsync("Error waiting for transcript.", response, ct)
+                .ConfigureAwait(false);
+            var jsonStream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
```

Comment:
better way with exception mapping
```suggestion
            var json = await response.Content.ReadAsStringWithExceptionMappingAsync().ConfigureAwait(false);
```